047-TrafficControlWithDapr/Student/Challenge-00.md

```diff
@@ -0,0 +1,356 @@
+### Architecture
+
+The traffic-control application architecture consists of four microservices:
+
+![Services](../images/services.png)
+
+- The **Camera Simulation** is a .NET Core console application that will simulate passing cars.
+- The **Traffic Control Service** is an ASP.NET Core WebAPI application that offers entry and exit endpoints: `/entrycam` and `/exitcam`.
+- The **Fine Collection Service** is an ASP.NET Core WebAPI application that offers 1 endpoint: `/collectfine` for collecting fines.
+- The **Vehicle Registration Service** is an ASP.NET Core WebAPI application that offers 1 endpoint: `/getvehicleinfo/{license-number}` for retrieving vehicle and owner information of a vehicle.
+
+These services compose together to simulate a traffic control scenario.
+
+The following sequence diagram describes how the application works:
+
+<img src="../images/sequence.png" alt="Sequence diagram" style="zoom:67%;" />
+
+1. The Camera `Simulation` generates a random license plate number and sends a *VehicleRegistered* message (containing this license plate number, a random entry-lane (1-3) and the timestamp) to the `/entrycam` endpoint of the `TrafficControlService`.
+1. The `TrafficControlService` stores the *VehicleState* (license plate number and entry-timestamp).
+1. After a random interval, the Camera `Simulation` sends a follow-up *VehicleRegistered* message to the `/exitcam` endpoint of the `TrafficControlService`. It contains the license plate number generated in step 1, a random exit-lane (1-3), and the exit timestamp.
+1. The `TrafficControlService` retrieves the previously-stored *VehicleState*.
+1. The `TrafficControlService` calculates the average speed of the vehicle using the entry and exit timestamp. It also stores the *VehicleState* with the exit timestamp for audit purposes, which is left out of the sequence diagram for clarity.
+1. If the average speed is above the speed-limit, the `TrafficControlService` calls the `/collectfine` endpoint of the `FineCollectionService`. The request payload will be a *SpeedingViolation* containing the license plate number of the vehicle, the identifier of the road, the speeding-violation in KMh, and the timestamp of the violation.
+1. The `FineCollectionService` calculates the fine for the speeding-violation.
+1. The `FineCollectionService` calls the `/vehicleinfo/{license-number}` endpoint of the `VehicleRegistrationService` with the license plate number of the speeding vehicle to retrieve vehicle and owner information.
+1. The `FineCollectionService` sends a fine notice to the owner of the vehicle by email.
+
+All actions described in the previous sequence are logged to the console during execution so you can follow the flow.
+
+The `Resources` folder contains the starting projects for the WhatTheHack. It contains a version of the services that use plain HTTP communication and store state in memory. With each challenge, you'll add a Dapr building block to enhance this application architecture.
```

Comment:
This should be modified to:

"You coach will provide you with a "Resources.zip" package file that contains the starting projects for this hack."

All subsequent mentions should refer to relative paths in "the 'Resource.zip' file provided by your coach".

I'm going to propose the entire sections referring to the traffic app's architecture be moved to a separate MD file/page.  This line about where to find the sample app & other resources should STAY in this Challenge-00.md file, but perhaps moved to the top of the pre-requisites section.  (Since obtaining the "resources.zip" file from your coach should be something done right away.)